ew_agents/data_eng_tools.py

```
@@ -136,49 +137,503 @@ def process_csv_data(csv_content: str, expected_columns: list = []) -> dict:
+        if "user" in column_mapping and column_mapping["user"] in row:
+            user = row[column_mapping["user"]]
+        if "user_id" in column_mapping and column_mapping["user_id"] in row:
+            user_id = row[column_mapping["user_id"]]
+        
+        # Extract engagement metrics
+        engagement = 0
+        if "engagement" in column_mapping and column_mapping["engagement"] in row:
+            try:
+                engagement = int(row[column_mapping["engagement"]].replace(",", ""))
+            except:
+                pass
```

Ruff (0.12.2)

Use `contextlib.suppress(BaseException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(BaseException)`

(SIM105)